packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,230 @@
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logRetention: logs.RetentionDays.ONE_WEEK,
+    });
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('Custom::LogRetention', {
+      RetentionInDays: customResourceLogRetention,
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+});
+
+describe('when a singleton-backed custom resource log group is specified', () => {
+  test('addLogRetentionLifetime modifies the retention period of a singleton-backed custom resource log group.', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup', {
+        retention: logs.RetentionDays.ONE_WEEK,
+      }),
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+
+  test('addLogRetentionLifetime only modifies the singleton-backed custom resource log group', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup1', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup1', {
+        retention: logs.RetentionDays.ONE_WEEK,
+      }),
+    });
+    new s3deploy.BucketDeployment(stack, 's3deployLogGroup2', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logGroup: new logs.LogGroup(stack, 'LogGroup2', {
+        retention: logs.RetentionDays.ONE_WEEK,
+      }),
+    });
+
+    const template = Template.fromStack(stack);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      LoggingConfig: {
+        LogGroup: {
+          Ref: 'LogGroup106AAD846',
+        },
+      },
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 2);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
```

Comment:
why is one log group retention supposed to be set, but not another? They're both bucket deployments, and they both have the explicitly created log group.

Reply:
One singleton (lambda) function from 2 BucketDeployment, that one singleton (lambda) only has one log group, the other log group is ignored as it's not attached to a singleton lambda.

Reply:
This is an interesting edge case, but this doesn't feel like behavior we want to enforce, because it's confusing and not entirely desirable. Let's remove this test case.